bridge/github/import_test.go

```diff
@@ -0,0 +1,223 @@
+	}
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	repo, err := repository.NewGitRepo(cwd, bug.Witnesser)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	backend, err := cache.NewRepoCache(repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer backend.Close()
+	interrupt.RegisterCleaner(backend.Close)
+
+	importer := &githubImporter{}
+	err = importer.Init(core.Configuration{
+		"user":    "MichaelMure",
+		"project": "git-but-test-github-bridge",
+		"token":   os.Getenv("GITHUB_TOKEN"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = importer.ImportAll(backend, time.Time{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ids := backend.AllBugsIds()
+	assert.Equal(t, len(ids), 8)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := backend.ResolveBugCreateMetadata(keyGithubUrl, tt.url)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			ops := b.Snapshot().Operations
+			if tt.exist {
+				assert.Equal(t, len(tt.bug.Operations), len(b.Snapshot().Operations))
+
+				for i, op := range tt.bug.Operations {
+					switch op.(type) {
+					case *bug.CreateOperation:
+						if op2, ok := ops[i].(*bug.CreateOperation); ok {
+							assert.Equal(t, op2.Title, op.(*bug.CreateOperation).Title)
+							assert.Equal(t, op2.Message, op.(*bug.CreateOperation).Message)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = CreationOperation", i)
+					case *bug.SetStatusOperation:
+						if op2, ok := ops[i].(*bug.SetStatusOperation); ok {
+							assert.Equal(t, op2.Status, op.(*bug.SetStatusOperation).Status)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = SetStatusOperation", i)
+					case *bug.SetTitleOperation:
+						if op2, ok := ops[i].(*bug.SetTitleOperation); ok {
+							assert.Equal(t, op.(*bug.SetTitleOperation).Was, op2.Was)
+							assert.Equal(t, op.(*bug.SetTitleOperation).Title, op2.Title)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = SetTitleOperation", i)
+					case *bug.LabelChangeOperation:
+						if op2, ok := ops[i].(*bug.LabelChangeOperation); ok {
+							assert.ElementsMatch(t, op.(*bug.LabelChangeOperation).Added, op2.Added)
+							assert.ElementsMatch(t, op.(*bug.LabelChangeOperation).Removed, op2.Removed)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = ChangeLabelOperation", i)
+					case *bug.AddCommentOperation:
+						if op2, ok := ops[i].(*bug.AddCommentOperation); ok {
+							assert.Equal(t, op.(*bug.AddCommentOperation).Message, op2.Message)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = AddCommentOperation", i)
+					case *bug.EditCommentOperation:
+						if op2, ok := ops[i].(*bug.EditCommentOperation); ok {
+							assert.Equal(t, op.(*bug.EditCommentOperation).Message, op2.Message)
+							continue
+						}
+						t.Errorf("bad operation type index = %d expected = EditCommentOperation", i)
+					default:
+
```

Comment:
default case should panic because we are not checking anything (that's a dev error, let's crash violently)